### PR TITLE
Update Go dependencies and minimal Go version to 1.20

### PR DESCRIPTION
**Pull request by lafriks**
I did not update docker dependencies to v23.0.1 as I did not know implications of that

Merge in changes and replaces #1628

### CODE REVIEW

**Review by 6543 — APPROVED** [review]

File: pipeline/stepBuilder.go [hunk 1]
```diff
@@ -279,9 +279,9 @@ func (b *StepBuilder) toInternalRepresentation(parsed *yaml.Config, environ map[
 		compiler.WithSecret(secrets...),
 		compiler.WithPrefix(
 			fmt.Sprintf(
-				"wp_%d_%d",
+				"wp_%s_%d",
```
Comment (lafriks) [inline pipeline/stepBuilder.go:282]: @6543 I changed to ULID so that volumes could be easily sortable by age

Comment (6543) [reply pipeline/stepBuilder.go:282]: oh not bad

Comment (6543) [reply pipeline/stepBuilder.go:282]: I did saw that you changed the lib and it looked nice but I did not notice that property

**Comment by lafriks** [comment]
Ok, I have no idea why migration tests are failing 😕

**Comment by 6543** [comment]
well that's a special case where the DBMS do not clean up the old index after table drop ... before we do a create table again ...

**Comment by 6543** [comment]
on old golang it just happend 1 in 100 cases ... so I just restarted pipeline if it happens but now it looks like we need a better solution

**Comment by lafriks** [comment]
@6543 what do you think about disabling that test for postgres and mysql until we find solution to make it stable?

**Comment by 6543** [comment]
I'm against it as we dont have to rush to get go1.20

**Comment by 6543** [comment]
and we would have missed e.g. https://github.com/woodpecker-ci/woodpecker/pull/1352/commits/7030cc19749771d4ca0b9b1dc45f55096a4d5920 ...

**Comment by lafriks** [comment]
> and we would have missed e.g. [7030cc1](https://github.com/woodpecker-ci/woodpecker/commit/7030cc19749771d4ca0b9b1dc45f55096a4d5920) (#1352) ...

wouldn't that be catch by sqlite migration test?

**Comment by lafriks** [comment]
Looks to me this should be good to go now

**Review by 6543 — APPROVED** [review]

**Pull request merged**